src/engine/client/client.cpp

```
@@ -203,6 +270,14 @@ void CClient::RconAuth(const char *pName, const char *pPassword)
 	if(pPassword != m_aRconPassword)
 		str_copy(m_aRconPassword, pPassword);
 
+	if(IsSixup())
+	{
+		CMsgPacker Msg7(protocol7::NETMSG_RCON_AUTH, true, true);
+		Msg7.AddString(pPassword, 32);
```

Is 0.7 rcon password length limited to 32? Arguments should probably be removed so the strings are sent entirely (see 0.6 code below).

---

The 32 is from here

https://github.com/teeworlds/teeworlds/blob/a1911c8f7d8458fb4076ef8e7651e8ef5e91ab3e/src/engine/client/client.cpp#L369

---

I think always sending the entire string makes more sense instead of truncating it. Could be removed for future proofing (i.e. if 0.7 would ever support longer passwords).

---

I like future proof but it might break login binds. Not too sure on that one. The password is compared with the config variable that also has a max size of 32 https://github.com/teeworlds/teeworlds/blob/a1911c8f7d8458fb4076ef8e7651e8ef5e91ab3e/src/engine/shared/config_variables.h#L94

Currently users can have a 512 long password set in both the client bind config and in the server side autoexec. It will be truncated to 32 on both ends and then match. If we extend the client side it might be annoying for users to debug why their authentication does not work. Using the same bind that worked for them in the vanilla client.

Its tricky but I don't like the trade off given the chances of how likely official teeworlds servers will have a 0.7.x not 0.8.x release with longer password support. Given the current upstream situation.

---

Truncating passwords has a bad history. We shouldn't pretend that we accept the longer password and truncate it to a "good size" so that the "authentication works". Let it blow up in the user's face so that they understand that their password was truncated behind their backs. Otherwise you'll have them think their long password actually works.

---

Okay I removed the limit. I still don't like it but fine.